Microsoft.Toolkit/Collections/ObservableGroupedCollectionExtensions.cs

```diff
@@ -33,8 +47,34 @@ public static class ObservableGroupedCollectionExtensions
         /// <param name="source">The source <see cref="ObservableGroupedCollection{TKey, TValue}"/> instance.</param>
         /// <param name="key">The key of the group to query.</param>
         /// <returns>The first group matching <paramref name="key"/> or null.</returns>
-        public static ObservableGroup<TKey, TValue> FirstOrDefault<TKey, TValue>(this ObservableGroupedCollection<TKey, TValue> source, TKey key)
-            => source.FirstOrDefault(group => GroupKeyPredicate(group, key));
+        [Pure]
+        public static ObservableGroup<TKey, TValue>? FirstOrDefault<TKey, TValue>(this ObservableGroupedCollection<TKey, TValue> source, TKey key)
+        {
+            if (source.TryGetList(out var list))
+            {
+                foreach (var group in list!)
+                {
+                    if (EqualityComparer<TKey>.Default.Equals(group.Key, key))
+                    {
+                        return group;
+                    }
+                }
+
+                return null;
+            }
+
+            return FirstOrDefaultWithLinq(source, key);
+        }
+
+        /// <summary>
+        /// Slow path for <see cref="First{TKey,TValue}"/>.
+        /// </summary>
+        [Pure]
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static ObservableGroup<TKey, TValue>? FirstOrDefaultWithLinq<TKey, TValue>(ObservableGroupedCollection<TKey, TValue> source, TKey key)
+        {
+            return source.FirstOrDefault(group => EqualityComparer<TKey>.Default.Equals(group.Key, key));
```

Comment:
nit: It can be a one liner

Reply:
Done in https://github.com/windows-toolkit/WindowsCommunityToolkit/pull/3260/commits/1a88b0bf04aec6e62dc3e1caec69a24cbbe0dfee.